app/controllers/media/objects_controller.rb

```diff
@@ -29,4 +45,17 @@ def media_filename
     extension = File.extname(@media_object.item_file_name)
     "elmo-#{@response.id}-#{@answer.id}#{extension}"
   end
+
+  def media_class(type)
+    case type
+    when 'audio'
+      return Media::Audio
+    when 'video'
+      return Media::Video
+    when 'image'
+      return Media::Image
+    else
+      return Media::Object
```

Comment:
Ok let's do that. I generally prefer failing loudly when something that
shouldn't happen does.

On Thu, Mar 17, 2016 at 7:59 PM, Melody <notifications@github.com> wrote:

> In app/controllers/media/objects_controller.rb
> <https://github.com/thecartercenter/elmo/pull/187#discussion_r56597010>:
>
> > @@ -29,4 +45,17 @@ def media_filename
> >      extension = File.extname(@media_object.item_file_name)
> >      "elmo-#{@response.id}-#{@answer.id}#{extension}"
> >    end
> > +
> > +  def media_class(type)
> > +    case type
> > +    when 'audio'
> > +      return Media::Audio
> > +    when 'video'
> > +      return Media::Video
> > +    when 'image'
> > +      return Media::Image
> > +    else
> > +      return Media::Object
>
> It shouldn't ever happen, the type comes from the URL params and the
> scope restricts to the valid types, so that condition should be
> more-or-less unreachable entirely. Raising an error there wouldn't be a bad
> idea.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/thecartercenter/elmo/pull/187/files/f4cd545baffd958d5a3e0f390a5944ca0edb251f#r56597010>
>



-- 
Tom Smyth

Worker-Owner, Sassafras Tech Collective
Specializing in innovative, usable tech for social change
sassafras.coop *·* @sassafrastech

Resident, Touchstone Cohousing
touchstonecohousing.org